### PR TITLE
network: Move common functions of bonding to bond-util

### PR DESCRIPTION
**Pull request by ssahani**
follow up https://github.com/systemd/systemd/pull/15152

### CODE REVIEW

**Comment by ssahani** [comment]
cc @yuwata

File: src/shared/bond-util.h [hunk 1]
```diff
@@ -0,0 +1,107 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+#pragma once
+
+#include <netinet/in.h>
+#include <linux/if_bonding.h>
+
+#include "in-addr-util.h"
+#include "ordered-set.h"
```
Comment (yuwata) [inline src/shared/bond-util.h]: These are not necessary.

File: src/shared/bond-util.c [hunk 2]
```diff
@@ -0,0 +1,78 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include "alloc-util.h"
+#include "bond-util.h"
+#include "conf-parser.h"
+#include "ether-addr-util.h"
+#include "extract-word.h"
+#include "string-table.h"
+#include "string-util.h"
```
Comment (yuwata) [inline src/shared/bond-util.c]: Please drop headers except `bond-util.h` and `string-table.h`.

File: src/network/netdev/bond.c [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 #include "alloc-util.h"
 #include "bond.h"
+#include "bond-util.h"
```
Comment (yuwata) [inline src/network/netdev/bond.c:5]: I guess `string-table.h` is not necessary any more.

**Comment by yuwata** [comment]
Please drop unnecessary headers in the first commit...

**Comment by ssahani** [comment]
Updated thanks @yuwata

File: src/shared/bond-util.h [hunk 4]
```diff
@@ -0,0 +1,106 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+#pragma once
+
+#include <netinet/in.h>
+#include <linux/if_bonding.h>
+
+#include "in-addr-util.h"
```
Comment (yuwata) [inline src/shared/bond-util.h]: Why is this necessary?

Comment (ssahani) [reply src/shared/bond-util.h]: yes It's not compiling.

Comment (poettering) [reply src/shared/bond-util.h]: all you need is to include "macro.h", so that you get `_pure_` and `_const_`. If you include that instead of in-addr-util.h all should be good.

**Comment by poettering** [comment]
looks superficially goo. But the `"in-addr-util.h"` inclusion in bond-util.h looks wrong, see above.

Also CI is unhappy.

File: src/shared/bond-util.h [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
```
Comment (poettering) [inline src/shared/bond-util.h:8]: in-addr-util.h is still included here, should be macro.h

Comment (ssahani) [reply src/shared/bond-util.h:8]: oops sry ! I missed it.

**Comment by ssahani** [comment]
Updated thanks for the review . PTAL.

File: src/network/networkctl.c [hunk 6]
```diff
@@ -1531,22 +1532,10 @@ static int link_status_one(
...
-                        { "802.3ad" },
-                        { "balance-tlb" },
-                        { "balance-alb" },
-                };
-
                 r = table_add_many(table,
                                    TABLE_EMPTY,
                                    TABLE_STRING, "Mode:",
-                                   TABLE_STRING, mode_table[info->mode],
+                                   TABLE_STRING,  bond_mode_to_string(info->mode),
```
Comment (keszybz) [inline src/network/networkctl.c:1538]: Extra space here, but ok.

**Comment by keszybz** [comment]
bionic-ppc64el failed with FAILED TESTS:  TEST-36-NUMAPOLICY, but it seems to be just a timeout?
```
May 17 10:43:14 systemd-testsuite systemd[1]: systemd-logind.service: start operation timed out. Terminating.
May 17 10:43:16 systemd-testsuite systemd[1]: systemd-logind.service: Failed with result 'timeout'.
May 17 10:43:16 systemd-testsuite systemd[1]: Failed to start systemd-logind.service.
May 17 10:43:18 systemd-testsuite systemd[1]: Starting of proc-sys-fs-binfmt_misc.automount not supported.
May 17 10:43:18 systemd-testsuite systemd[1]: testsuite-36.service: Failed with result 'exit-code'.
May 17 10:43:19 systemd-testsuite systemd[1]: Failed to start testsuite-36.service.
```
